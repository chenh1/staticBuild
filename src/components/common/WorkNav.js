import React, {PropTypes} from 'react';
import { Link, IndexLink } from 'react-router';

class WorkNav extends React.Component{
    render() {
        return(
            <div className="work-content">
                <a data-title="AMMA: An asthma monitoring an management app" className="amma portfolio-item page-link" href="/amma/">
                    <div className="image">
                        <img src={require("../../assets/amma-main.png")} alt="AMMA"></img>
                    </div>
                    <div className="info">
                        <h3 className="header-three">AMMA</h3>
                        <p className="paragraph-one">
                            Helping asthma sufferers manage and monitor their illness day to day
                        </p>
                    </div>
                </a>

                <a data-title="Dignity Meals" className="dignity-meals portfolio-item page-link" href="/dignity-meals/">
                    <div className="image">
                        <img src={require("../../assets/dignity-main.png")} alt="Dignity Meals"></img>
                    </div>
                    <div className="info">
                        <h3 className="header-three">Dignity Meals</h3>
                        <p className="paragraph-one">
                            A food rescue and redistribution app focused on reducing hunger and food waste
                        </p>
                    </div>
                </a>

                <a data-title="Abundant Life" className="abundant-life portfolio-item page-link" href="/abundant-life/">
                    <div className="image">
                        <img src={require("../../assets/alife.png")} alt="Abundant Life"></img>
                    </div>
                    <div className="info">
                        <h3 className="header-three">Abundant Life</h3>
                        <p className="paragraph-one">
                            Coming soon!
                        </p>
                    </div>
                </a>

                <a data-link-type="externalLink" data-title="Graphic Design" className="graphic-design portfolio-item page-link" href="http://cargocollective.com/tracylin">
                    <div className="image">
                        <img src={require("../../assets/graphic-design.png")} alt="Graphic Design"></img>
                    </div>
                    <div className="info">
                        <h3 className="header-three">Graphic Design</h3>
                        <p className="paragraph-one">
                            Select visual design projects that I've produced in the past years
                        </p>
                    </div>
                </a>
            </div>
        );
    }
}

export default WorkNav;