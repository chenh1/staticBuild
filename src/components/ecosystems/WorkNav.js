import React, {PropTypes} from 'react';
import { WorkItem } from '../organisms';

const WorkNav = ({workItems}) => {
    return (
        <div id="workContent" className="work-content">
            {workItems.map((workItem, index) => {
                return (
                    <WorkItem 
                        key={'workItem' + index}
                        path={workItem.path}
                        customClass={workItem.customClass}
                        imageSrc={workItem.imageSrc}
                        alt={workItem.alt}
                        subHeader={workItem.subHeader}
                        description={workItem.description}
                    />
                );
            })}
        </div>
    );
};

WorkNav.propTypes = {
    workItems: PropTypes.array.isRequired
};

export default WorkNav;