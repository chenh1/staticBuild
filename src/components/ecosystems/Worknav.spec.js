import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import WorkNav from './WorkNav';

test('renders WorkNav', () => {
    const tree = renderer.create(<WorkNav
        workItems={[
            {
                path: '/mockUrl', 
                customClass: 'mock-class', 
                imageSrc: '/mockSrc',
                alt: 'mock alt',
                subHeader: 'Mock sub-header text',
                description: ['Mock paragraph text']
            }
        ]} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});