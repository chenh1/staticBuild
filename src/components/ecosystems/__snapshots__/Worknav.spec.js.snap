// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders WorkNav 1`] = `
<div
  className="work-content"
  id="workContent"
>
  <a
    className="mock-class"
    onClick={[Function]}
    style={Object {}}
  >
    <div
      className="image"
    >
      <img
        alt="mock alt"
        src="/mockSrc"
      />
      
    </div>
    <div
      className="info"
    >
      <h3
        className="header-three"
      >
        Mock sub-header text
      </h3>
      <p
        className="paragraph-one"
      >
        Mock paragraph text
      </p>
    </div>
  </a>
</div>
`;
